src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -34,32 +42,38 @@ protected BaseSync(@Nonnull final U statistics, @Nonnull final V syncOptions) {
      * this list with it's corresponding old resource in a given CTP project, and in turn it either issues update
      * actions on the existing resource if it exists or create it if it doesn't.
      *
-     * <p>The time before and after the actual sync process starts is recorded in the {@link BaseSyncStatistics}
-     * container so that the total processing time is computed in the statistics.
-     *
      * @param resourceDrafts the list of new resources as drafts.
-     * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     * @return an instance of {@link CompletionStage&lt;S&gt;} which contains as a result an instance of {@link S} which
+     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} of this sync process.
      */
-    public CompletionStage<U> sync(@Nonnull final List<T> resourceDrafts) {
-        statistics.startTimer();
-        return process(resourceDrafts).thenApply(resultingStatistics -> {
-            resultingStatistics.calculateProcessingTime();
+    public CompletionStage<S> sync(@Nonnull final List<T> resourceDrafts) {
+        final long startTime = System.currentTimeMillis();
+        return process(resourceDrafts).thenApply(resultingStatisticsBuilder -> {
+            resultingStatisticsBuilder.setProcessingTimeInMillis(System.currentTimeMillis() - startTime);
+            final S resultingStatistics = resultingStatisticsBuilder.build();
+            updateTotalStatistics(resultingStatistics);
             return resultingStatistics;
         });
     }
 
 
     /**
-     * Returns an instance of type U which is a subclass of {@link BaseSyncStatistics} containing all the stats of the
-     * sync process; which includes a report message, the total number of update, created, failed, processed resources
-     * and the processing time of the sync in different time units and in a human readable format.
+     * Returns an instance of type S which is a subclass of {@link BaseSyncStatistics} containing the total stats of
+     * all sync processes performed by {@code this} sync instance.
      *
-     * @return a statistics object for the sync process.
+     * @return a statistics object for all sync processes performed by {@code this} sync instance.
      */
     @Nonnull
-    public U getStatistics() {
-        return statistics;
+    public synchronized S getStatistics() {
```

Comment:
@andrii-kovalenko-ct 
`getStatistics()` was there since the beginning, when `sync()` returned the `void`. Now `sync()` returns `CompletionStage` containing a distinct statistics instance for every subsequent invocations. Because of that, keeping `getStatistics()` makes sense only if we want to hold a merged statistics of all `sync()` invocations performed by a single `BaseSync` instance. The question is do we want to do so? 
If the answer is no, then I'll delete both `update/get` functions. The user will be able to merge statistics by himself. 
Otherwise I think that concurrent `updateTotalStatistics()` is possible, cause user may invoke`sync()` multiple times concurrently e.g. for different chunks fetched from CSV. That's why we need a synchronization here.